Sources/xcproj/PBXProjObjects+Helpers.swift

```diff
@@ -15,7 +16,160 @@ public extension PBXProj.Objects {
         targets.append(contentsOf: Array(aggregateTargets.values) as [PBXTarget])
         return targets.filter { $0.name == name }
     }
-    
+
+    /// Retruns target's sources build phase.
+    ///
+    /// - Parameter target: target object.
+    /// - Returns: target's sources build phase, if found.
+    public func sourcesBuildPhase(target: PBXTarget) -> PBXSourcesBuildPhase? {
+        return sourcesBuildPhases.first(where: { target.buildPhases.contains($0.key) })?.value
+    }
+
+    /// Returns all files in target's sources build phase.
+    ///
+    /// - Parameter target: target object.
+    /// - Returns: all files in target's sources build phase, or empty array if sources build phase is not found.
+    public func sourceFiles(target: PBXTarget) -> [PBXFileElement] {
+        return sourcesBuildPhase(target: target)?.files
+            .flatMap({ buildFiles[$0]?.fileRef })
+            .flatMap(getFileElement(reference:)) ?? []
+    }
+
+    /// Returns group with the given name contained in the given parent group and its reference.
+    ///
+    /// - Parameter groupName: group name.
+    /// - Parameter inGroup: parent group.
+    /// - Returns: group with the given name contained in the given parent group and its reference.
+    public func group(named groupName: String, inGroup: PBXGroup) -> (String, PBXGroup)? {
+        let children = inGroup.children
+        return groups.first(where: {
+            children.contains($0.key) && ($0.value.name == groupName || $0.value.path == groupName)
+        })
+    }
+
+    /// Adds new group with the give name to the given parent group.
+    /// Group name can be a path with components separated by `/`.
+    /// Will create new groups for intermediate path components or use existing groups.
+    /// Returns new group and its reference.
+    ///
+    /// - Parameters:
+    ///   - groupName: group name, can be a path with components separated by `/`
+    ///   - children: group children, default is empty array.
+    ///   - sourceTree: group source tree, default is `.group`.
+    ///   - name: group name.
+    ///   - path: group path.
+    ///   - usesTabs: group uses tabs.
+    ///   - options: additional options, default is empty set.
+    /// - Returns: new group and its reference.
+    public func addGroup(
+        named groupName: String,
```

Comment:
The problem that I see exposing this method with all these arguments is that if any property is introduced in the `PBXGroup` class, we need to update this method to support it. Given that xcproj represents the objects with mutable classes what do you think about simplifying it:

```swift
public func addGroup(named groupName: String,
  to toGroup: PBXGroup, 
  withFolder: Bool = false) -> [(String, PBXGroup)]
```
Notice that the method returns an array with all the groups that got created. If the developer passes a path like this `/group_a/group_b/group_c` then it'd return `[GroupA, GroupB, GroupC]` corresponding the last element in the array to the last path component. If the developer is interested in updating let's say, the `usesTabs` property of all the created groups, it can be easily done iterating with a `forEach` on the returned array.

On the other side, I think the implementation could be simplified adding some recursion:

```
// Given the path /component_a/component_b/component_c

1. Generate/Get the first component group: GroupA
2. Concatenate the group with the result of calling the method with GroupA and name /component_b/component_c: [GroupA, result[0], result[1], ...]
```

Reply:
Right, I didn't notice that those types are mutable classes, so it will makes sense to remove all those parameters. And I like your idea about returning array.

Reply:
As Xcode allows you to use `/` in group name, i think this should not be used as separator.

Reply:
Can you show an example? At least Xcode 9 does not allow to rename groups using `/`.

Reply:
Once you point a group to a folder with name different than group name, you can change group name, without changing the folder name. 
Example: https://imgur.com/oveZwsI

Reply:
Ok. This seems to be allowed also when group is created without folder. What is the alternative then? We can require passing in array of group names instead. But then if `/` is used in one of them we should not create a group with such name if it is created with folder, as it will probably lead to invalid project. We should probably normalise each item. But if it is created without folder we should allow that.